numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1738,12 +1738,30 @@ unpack_bits(PyObject *input, int axis)
     }
 
     /* Setup output shape */
-    for (i=0; i<PyArray_NDIM(new); i++) {
+    for (i = 0; i < PyArray_NDIM(new); i++) {
         outdims[i] = PyArray_DIM(new, i);
     }
 
     /* Multiply axis dimension by 8 */
-    outdims[axis] <<= 3;
+    outdims[axis] *= 8;
+    if (count_obj != Py_None) {
+        count = PyArray_PyIntAsIntp(count_obj);
```

Comment:
For review: `count_obj` is a borrowed reference up to this point. Is that going to be OK here?